### DISCUSSION TITLE
테스트에서만 필요한 메서드에 대해서 어떻게 생각하시나요?

### DISCUSSION THREAD

**Discussion by le2sky** [2024-woowacourse-study/level-interview#44] [Test]
다른 크루들의 의견이 필요한 부분입니다.

제 생각은 "만들어도 괜찮지만, 적당히 하기(기본적으로는 지양, 만들었을때 가치가 더 크면 만들기)" 입니다. 

이 질문에 답할 때, 테스트에서만 필요한 메서드를 만들어서 발생하는 문제점도 설명해주면 좋을듯?

이걸 보는 당신의 의견이 궁금해요.

**Comment by gyungchan-jo** [comment]
테스트를 위한 생성자는 개인적으로 좋다 생각합니다.
이유는 다수의 생성자를 제공하면 객체를 다양한 방식으로 초기화할 수 있습니다. 이는 클라이언트 코드가 객체를 생성할 때 필요한 초기화 옵션을 선택할 수 있는 유연성을 제공하기 때문입니다.

하지만 테스트에서만 필요한 메서드는 만들면 안된다 생각합니다.
아직 테스트를 위한 메서드가 필요한 경우가 없었기 때문에 문제점을 경험해보진 못했습니다.
그럼에도 생각해볼만한 문제점은 만약 다른 페어와 개발을 진행할 때, 나는 테스트만을 위해 작성한 메서드였지만 페어는 그 사실을 인지하지 못하고 해당 메서드를 그대로 사용할 수 있을 것 같고 이를 통해 예상치 못한 예외를 발생시킬 수 있다 생각이 됩니다.

**Comment by le2sky** [comment]
테스트에서만 필요한 무언가(getter, 생성자, 어쩌구)는 기본적으로 지양하는 것이 좋다고 생각합니다.
프로덕션 코드에서 필요하지 않은 코드를 작성하는 것은 다른 개발자에게 혼란을 줄 수 있습니다.
하지만, 만들었을 때 얻을 수 있는 이점이 명확하다면 만드는 것이 실용적이라고 생각합니다.

**Comment by GIVEN53** [comment]
getter는 지양
필요하기 전까지 생성을 미루는 편
캡슐화

생성자는 합리적일 경우 ok
다양한 생성 로직 제공 -> 유연성
